presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/QueryBuilder.java

```diff
@@ -138,6 +140,9 @@ else if (typeAndValue.getType().equals(TinyintType.TINYINT)) {
             else if (typeAndValue.getType().equals(DoubleType.DOUBLE)) {
                 statement.setDouble(i + 1, (double) typeAndValue.getValue());
             }
+            else if (typeAndValue.getType().equals(RealType.REAL)) {
+                statement.setFloat(i + 1, intBitsToFloat((int) ((long) typeAndValue.getValue())));
```

Comment:
Can we do something similar to what is being done for Tinyint instead of double casting?